src/graphql/data/TopTokens.ts

```diff
@@ -0,0 +1,222 @@
+    if (showFavorites) {
+      returnTokens = returnTokens?.filter((token) => (token?.address && favorites.includes(token.address)) ?? false)
+    }
+    if (lowercaseFilterString) {
+      returnTokens = returnTokens?.filter((token) => {
+        const addressIncludesFilterString = token?.address?.toLowerCase().includes(lowercaseFilterString)
+        const nameIncludesFilterString = token?.name?.toLowerCase().includes(lowercaseFilterString)
+        const symbolIncludesFilterString = token?.symbol?.toLowerCase().includes(lowercaseFilterString)
+        return (nameIncludesFilterString || symbolIncludesFilterString || addressIncludesFilterString) ?? false
+      })
+    }
+    return returnTokens
+  }, [tokens, showFavorites, lowercaseFilterString, favorites])
+}
+
+const PAGE_SIZE = 20
+//const MAX_COUNT = 100
+
+function toContractInput(token: PrefetchedTopToken) {
+  return {
+    address: token?.address ?? '',
+    chain: token?.chain ?? 'ETHEREUM',
+  }
+}
+
+export type TopToken = NonNullable<TopTokens_TokensQuery['response']['tokens']>[number]
+export function useTopTokens(prefetchedTokens: TopTokens100Query['response']) {
+  // TODO: add filtering by favorites and filter string
+  const duration = toHistoryDuration(useAtomValue(filterTimeAtom))
+  const environment = useRelayEnvironment()
+  const [tokens, setTokens] = useState<TopToken[]>()
+
+  const [page, setPage] = useState(1)
+  const selectedTokens = useFilteredTokens(useSortedTokens(prefetchedTokens.topTokens))
+  const [isFetching, setIsFetching] = useState(true)
+
+  // TopTokens should ideally be fetched with usePaginationFragment. The backend does not current support graphql cursors;
+  // in the meantime, fetchQuery is used, as other relay hooks do not allow the refreshing and lazy loading we need
+  const loadTokens = useCallback(
+    (contracts: ContractInput[], onSuccess: (data: TopTokens_TokensQuery['response'] | undefined) => void) => {
+      fetchQuery<TopTokens_TokensQuery>(
+        environment,
+        tokensQuery,
+        { contracts, duration },
+        { fetchPolicy: 'store-or-network' }
+      )
+        .toPromise()
+        .then(onSuccess)
+    },
+    [duration, environment]
+  )
+
+  const loadMoreTokens = () => {
+    const contracts = selectedTokens.slice(page * PAGE_SIZE, (page + 1) * PAGE_SIZE).map(toContractInput)
+    loadTokens(contracts, (data) => {
+      if (data?.tokens) {
+        setTokens([...(tokens ?? []), ...data.tokens])
+        setPage(page + 1)
+      }
+    })
+  }
+
+  // Reset count when filters are changed
+  useEffect(() => {
+    setIsFetching(true)
+    setTokens([])
+    const contracts = selectedTokens.slice(0, PAGE_SIZE).map(toContractInput)
+    loadTokens(contracts, (data) => {
+      if (data?.tokens) {
+        console.log('D')
+        setTokens([...data.tokens])
+        setIsFetching(false)
+        setPage(1)
+      }
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
if this is necessary, then you should add a comment explaining why the specific variables that are missing were excluded